extractor_api.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func NewExtractorAPI(extractor *Extractor) http.Handler {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/healthcheck", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.Header().Set("Allow", "GET")
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		w.Write([]byte("OK"))
+	})
+
+	mux.HandleFunc("/extract", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			w.Header().Set("Allow", "POST")
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		postBody := make([]byte, 10000)
+		r.Body.Read(postBody)
+		matchedTermIds := extractor.Extract(string(postBody))
```

Comment:
It would probably be more idiomatic to make `extractor.Extract` take an `io.Reader` here.  This would allow the function to read the input in a streamed fashion as opposed to buffering it all in memory.  It also makes the extractor more flexible as it can be given all sorts of things (eg an open file, a network stream, a [`strings.Reader`](http://golang.org/pkg/strings/#Reader) etc.)